src/Tribe/Views/V2/Messages.php

```diff
@@ -120,7 +120,26 @@ public static function for_key( $key, ...$values ) {
 		// If not found return the key itself.
 		$match = Arr::get( $map, $key, $key );
 
-		return count( $values ) ? sprintf( $match, ...$values ) : $match;
+		if ( empty( count( $values ) ) ) {
+			return $match;
+		}
+
+		$need_events_label_keys = [ 'day_no_results_found' ];
+
+		/**
+		 * Filters the array of keys of the messages that need the events label.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $need_events_label_keys Array of keys of the messages that need events label.
+		 */
+		$need_events_label_keys = apply_filters( 'tribe_events_views_v2_messages_need_events_label_keys', $need_events_label_keys );
+
+		if ( in_array( $key, $need_events_label_keys ) ) {
+			array_unshift( $values, tribe_get_event_label_plural() );
```

Comment:
not sure if this is the best way to do this, also see https://github.com/moderntribe/events-pro/pull/1512 for the counterpart to the filter.

Reply:
After a hangout about this, I think Paul's solution is a good balance between localization, white-labeling and maintainability concerns.